From e495cee1edaabe121fb1e55e28f7087e2bea338b Mon Sep 17 00:00:00 2001
From: Eric Sandeen <sandeen@redhat.com>
Date: Thu, 9 Oct 2014 21:00:14 -0400
Subject: [fs] xfs: keep sb_bad_features2 the same a sb_features2

Message-id: <1412888441-4181-21-git-send-email-sandeen@redhat.com>
Patchwork-id: 97317
O-Subject: [RHEL7.1 PATCH 20/47] [fs] xfs: keep sb_bad_features2 the same a sb_features2
Bugzilla: 1145837
RH-Acked-by: Brian Foster <bfoster@redhat.com>
RH-Acked-by: Dave Chinner <dchinner@redhat.com>

Bugzilla: 1145837
Upstream Status: Committed upstream
Build Info: https://brewweb.devel.redhat.com/taskinfo?taskID=8082426
Tested: locally using xfstests

Thanks,
-Eric

    commit 32bf1deae15b67839a2f0cb233254125be73fc78
    Author: Dave Chinner <dchinner@redhat.com>
    Date:   Tue May 20 07:41:43 2014 +1000

    xfs: keep sb_bad_features2 the same a sb_features2

    Whenever we update sb_features2, we need to update sb_bad_features2
    so that they remain identical on disk. This prevents future mounts
    or userspace utilities from getting confused over which features the
    filesystem supports.

    Signed-off-by: Dave Chinner <dchinner@redhat.com>
    Reviewed-by: Christoph Hellwig <hch@lst.de>
    Signed-off-by: Dave Chinner <david@fromorbit.com>
Signed-off-by: Eric Sandeen <sandeen@redhat.com>
---

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/fs/xfs/xfs_sb.h b/fs/xfs/xfs_sb.h
index f7b2fe7..f350705 100644
--- a/fs/xfs/xfs_sb.h
+++ b/fs/xfs/xfs_sb.h
@@ -536,11 +536,13 @@ static inline void xfs_sb_version_addattr2(xfs_sb_t *sbp)
 {
  sbp->sb_versionnum |= XFS_SB_VERSION_MOREBITSBIT;
  sbp->sb_features2 |= XFS_SB_VERSION2_ATTR2BIT;
+ sbp->sb_bad_features2 |= XFS_SB_VERSION2_ATTR2BIT;
 }
 
 static inline void xfs_sb_version_removeattr2(xfs_sb_t *sbp)
 {
  sbp->sb_features2 &= ~XFS_SB_VERSION2_ATTR2BIT;
+ sbp->sb_bad_features2 &= ~XFS_SB_VERSION2_ATTR2BIT;
  if (!sbp->sb_features2)
   sbp->sb_versionnum &= ~XFS_SB_VERSION_MOREBITSBIT;
 }
-- 
1.7.1